geomagio/WebService.py

```diff
@@ -81,7 +78,77 @@ def parse(params):
         WebServiceQuery
             parsed query object.
         """
-        pass
+        # Create dictionary of lists
+        dict = parse_qs(params)
+        # Return values
+        id = dict.get('id', [''])[0]
+        starttime = dict.get('starttime', [''])[0]
+        endtime = dict.get('endtime', [''])[0]
+        elements = dict.get('elements', [''])[0]
+        sampling_period = dict.get('sampling_period', [''])[0]
+        type = dict.get('type', [''])[0]
+        format = dict.get('format', [''])[0]
+        # Escape to avoid script injection
+        self.id = escape(id)
+        self.starttime = escape(starttime)
+        self.endtime = escape(endtime)
+        elements = escape(elements)
+        self.sampling_period = escape(sampling_period)
+        self.type = escape(type)
+        self.format = escape(format)
+        # Check for values
+        if not self.id:
+            self.id = 'BOU'
+        if not starttime or not endtime:
+            self.starttime = str(datetime.utcnow().strftime("%Y-%m-%dT")) + '00:00:00Z'
```

Comment:
- when starttime is not specified it should default to the start of the current day.
- when endtime is not specified it should default to to starttime + 24 hours
- Convert to UTCDateTime objects during parsing, rather than during fetch